Makefile

```diff
@@ -307,30 +306,39 @@ functional-with-fault-injection-test: clean-test-results
 
 test: unit-test integration-test functional-test functional-with-fault-injection-test
 
-##### Coverage #####
+##### Coverage & Reporting #####
 $(COVER_ROOT):
 	@mkdir -p $(COVER_ROOT)
 
-unit-test-coverage: $(COVER_ROOT)
+$(REPORT_ROOT):
+	@mkdir -p $(REPORT_ROOT)
+
+prepare-coverage-test: update-gotestsum $(COVER_ROOT) $(REPORT_ROOT)
+
+unit-test-coverage: prepare-coverage-test
 	@printf $(COLOR) "Run unit tests with coverage..."
-	@echo "mode: atomic" > $(UNIT_COVER_PROFILE)
```

Comment:
Removed this line. I don't think it's needed because the task to create the summary also does it. Checking a [build for this branch in Coveralls](https://coveralls.io/builds/63115398), the coverage looks to be reported correctly.

Reply:
Yup, also deleted in my branch